src/main/java/com/socialmediaraiser/twitterbot/impl/personalAnalyzer/PersonalAnalyzerBot.java

```diff
@@ -1,130 +1,175 @@
+          this.ioHelper.addNewFollowerLineSimple(usersToWrite);
+          usersToWrite = new ArrayList<>();
+          LOGGER.info("adding " + nbUsersToAdd + " users ...");
+          TimeUnit.MILLISECONDS.sleep(500);
         }
+      }
+    }
+    this.ioHelper.addNewFollowerLineSimple(usersToWrite);
+    LOGGER.info("finish with success");
+  }
+
+  private boolean hasToAddUser(IUser user, List<IUser> followings, List<IUser> followers,
+                               boolean showFollowings, boolean showFollowers, boolean onyFollowBackUsers) {
+    // case 0 : only follow back users
+    if (onyFollowBackUsers && followings.contains(user) && !followers.contains(user)) {
+      return false;
     }
 
-    private UserInteractions getNbInterractions(String archiveFileName) throws IOException {
-        File file = new File(getClass().getClassLoader().getResource(archiveFileName).getFile());
-        UserInteractions userInteractions = new UserInteractions();
-        // counts all retweets given to others
-        dataArchiveHelper.countGivenRetweets(userInteractions);
-        // counts all the unique replies given by the user to others
-        dataArchiveHelper.countRepliesGiven(userInteractions);
-        // counts all the retweets of user tweets done by others
-        dataArchiveHelper.countRetweesReceived(userInteractions);
-        // counts all replies given recently to others
-        apiSearchHelper.countRecentRepliesGiven(userInteractions, dataArchiveHelper.filterTweetsByRetweet(false).get(0).getCreatedAt()); // @todo test 2nd arg
-        // counts all the replies received by others
-        apiSearchHelper.countRepliesReceived(userInteractions, true); // D-7 -> D0
-        apiSearchHelper.countRepliesReceived(userInteractions, false); // D-30 -> D-7
-        apiSearchHelper.countGivenLikesOnStatuses(userInteractions);
-        return userInteractions;
+    // case 1 : show all the people i'm following and all the users following me
+    if (!showFollowers && !showFollowings) {
+      return true;
     }
+    // case 2 : show all the people I'm following who are following me back
+    else if (showFollowers && showFollowings && onyFollowBackUsers) {
+      return (followings.contains(user) && followers.contains(user));
+    }
+    // case 3 : show all the people i'm following or all the people who are following me
+    else {
+      return ((followings.contains(user) && showFollowings) || followers.contains(user) && showFollowers);
+    }
+  }
 
+  private Map<String, UserStats> getNbInteractions() {
+    Stream<Tuple2<String, UserInteraction>> givenInteractions = this.getGivenInteractions().toStream();
+    var result = givenInteractions
+        .peek(ui -> LOGGER.info("analyzing : " + ui._1()))
+        .groupBy(Tuple2::_1)
+        .map(ui -> buildTurpleFromUserInteractions(ui._1(), ui._2()));
+    return result; // @todo KO : is not Map<String, UserStats> but Seq<Object> instead !
```

Comment:
@hypr2771 if you have an idea about why the type doesn't match... I tried everything without success :(

Reply:
Edit: I think I get it, had to add `HashMap.ofEntries(` before 👯